### PR TITLE
Fix genconfig does not fail when schema validation failed

### PR DESCRIPTION
**Pull request by blackchoey**
Fix genconfig does not fail when schema validation failed #424 
AB#1696447

### CODE REVIEW

File: iotedgedev/deploymentmanifest.py [hunk 1]
```diff
@@ -190,6 +190,9 @@ def _validate_json_schema(self, schema_object, json_object, schema_type):
                 self.output.warning("%s schema validation failed. Please see previous logs for more details" % schema_type)
             else:
                 self.output.info("%s schema validation passed." % schema_type)
+
+            if error_detected:
```
Comment (adashen) [inline iotedgedev/deploymentmanifest.py]: line 189 has if error_detected?

Comment (adashen) [reply iotedgedev/deploymentmanifest.py]: besides it already has error_detected, do we still need validation_success?

File: iotedgedev/deploymentmanifest.py [hunk 2]
```diff
@@ -146,7 +146,7 @@ def validate_deployment_template(self):
         validation_success = True
         try:
             template_schema = json.loads(urlopen(Constants.deployment_template_schema_url).read().decode())
-            self._validate_json_schema(template_schema, self.json, "Deployment template")
+            validation_success = self._validate_json_schema(template_schema, self.json, "Deployment template")
         except Exception as ex:  # Ignore other non shcema validation errors
             self.output.info("Unexpected error during deployment template schema validation, skip schema validation. Error:%s" % ex)
```
Comment (adashen) [inline iotedgedev/deploymentmanifest.py]: exception happens and the validation is still success?

File: iotedgedev/deploymentmanifest.py (hunk 1 above)
Comment (blackchoey) [reply iotedgedev/deploymentmanifest.py]: Removed the error_detected and reuse validation_success. We need to declare validation_success outside of try to handle situations with exception.

File: iotedgedev/deploymentmanifest.py (hunk 2 above)
Comment (blackchoey) [reply iotedgedev/deploymentmanifest.py]: Only schema failures should be treated as validation failure. Other exceptions such as network error should not fail the validation. iotedgedev will only print related error and skip validation.

File: iotedgedev/deploymentmanifest.py (hunk 1 above)
Comment (adashen) [reply iotedgedev/deploymentmanifest.py]: removed error_detected? we have two blocks of error_detected (189 and 194)?

**Comment by blackchoey** [comment]
Some Azure CLI modules are broken, I will investigate and see whether I can provide a quick fix

File: iotedgedev/deploymentmanifest.py (hunk 1 above)
Comment (blackchoey) [reply iotedgedev/deploymentmanifest.py]: Here's the commit that removes error_detected, I removed both of them and use validation_success instead: https://github.com/Azure/iotedgedev/pull/425/commits/60cac6655901b22fad8de6336616c1e28e042373

**Pull request merged**